src/Stencil.php

```diff
@@ -0,0 +1,294 @@
+    public function getPicturesFields(bool $editor): array
+    {
+        return [];
+    }
+
+    public function getParams(bool $editor): array
+    {
+        return [];
+    }
+
+    public function addNewZones(array $input)
+    {
+        $this->update(array_merge($input, [
+            'nb_zones' => $this->fields['nb_zones'] + ($input['nb-new-zones'] ?? 1),
+        ]));
+    }
+
+    public function removeZones(array $input)
+    {
+        // Compute the new number of zones
+        $nbZones = $this->fields['nb_zones'] - ($input['nb-new-zones'] ?? 1);
+
+        // Remove zones with id > $nbZones
+        $zones = array_filter(
+            json_decode($this->fields['zones'] ?? '{}', true),
+            fn ($id) => $id <= $nbZones,
+            ARRAY_FILTER_USE_KEY
+        );
+
+        // Update the stencil
+        $this->update(array_merge($input, [
+            'zones'    => json_encode($zones, JSON_FORCE_OBJECT),
+            'nb_zones' => $this->fields['nb_zones'] - ($input['nb-new-zones'] ?? 1),
+        ]));
+    }
+
+    /**
+     * Load the cropper library
+     *
+     */
+    public static function loadLibs(): void
+    {
+        echo Html::script("public/lib/cropper.js");
+        echo Html::script("js/stencil-editor.js");
+        echo Html::scss('css/standalone/stencil-editor.scss');
+    }
+
+    public function getTabNameForItem(CommonGLPI $item, $withtemplate = 0): string
+    {
+        $nb = count(json_decode($this->getStencilFromItem($item)->fields['zones'] ?? '{}', true));
+
+        return self::createTabEntry($this->getTypeName(), $nb, $item::getType());
+    }
+
+    public function displayStencil(): void
+    {
+        $item = $this->getStencilItem();
+
+        if ($item == null) {
+            return;
+        }
+
+        self::loadLibs();
+
+        $pictures = [];
+        foreach ($this->getPicturesFields(false) as $picture_field) {
+            if (
+                empty($item->getItemtypeOrModelPicture($picture_field))
+                || !isset($item->getItemtypeOrModelPicture($picture_field)[0]['src'])
+            ) {
+                continue;
+            }
+
+            $pictures[] = $item->getItemtypeOrModelPicture($picture_field)[0]['src'];
+        }
+
+        $model_class   = $item->getModelClass();
+        $model_fk      = $item->getModelForeignKeyField();
+        $model         = new $model_class();
+        $model->getFromDB($item->fields[$model_fk]);
+        $model_stencil = self::getStencilFromItem($model);
+
+        TemplateRenderer::getInstance()->display('stencil/view.html.twig', [
+            'item'              => $item,
+            'stencil'           => $this,
+            'zones'             => json_decode($model_stencil->fields['zones'] ?? '{}', true),
+            'pictures'          => $pictures,
+            'params'            => array_merge(
+                $this->getParams(false),
+                ['is_editor_view' => false]
+            ),
+        ]);
+    }
```

Comment:
Lots of warning because of `$item` being a `CommonGLPI`:

![image](https://github.com/glpi-project/glpi/assets/42734840/3f9f0dbd-883b-4ba1-8550-364dc960556e)

Shouldn't `getStencilItem` return `CommonDBTM` items instead ?